src/main/java/org/wildfly/security/ssl/SSLUtils.java

```diff
@@ -127,12 +127,13 @@ private SSLUtils() {}
                             } catch (NoSuchAlgorithmException ignored) {}
                         }
                 }
-                throw ElytronMessages.log.noAlgorithmForSslProtocol();
-            };
-        }
 
-        if (log.isTraceEnabled()) {
```

Comment:
I think we should also still leave this ```if``` statement as is - if we get here, it means that there were no supported protocols to try (which is a different case from the above ```if``` statement, where there are supported protocols to try).

Reply:
It would be good to add trace logging to indicate the supported protocols that are determined on line 119.

Reply:
It would also be good to add trace logging to indicate the providers that support ```SSLContext``` - e.g., after line 112, you could add trace logging to indicate the algorithm name and provider name that are added to the map.

Reply:
Fixed/added